File: src/Sample/Flutter.Net/MaterialButtonsDemoPage.cs
using System;
using Avalonia;
using Avalonia.Media;
using Flutter.Material;
using Flutter.Rendering;
using Flutter.Widgets;

// Dart parity source (reference): dart_sample/lib/material_buttons_demo_page.dart (exact sample parity)

namespace Flutter.Net;

public sealed class MaterialButtonsDemoPage : StatefulWidget
{
    public override State CreateState()
    {
        return new MaterialButtonsDemoPageState();
    }
}

internal sealed class MaterialButtonsDemoPageState : State
{
    private bool _enabled = true;
    private int _textButtonTaps;
    private int _elevatedButtonTaps;
    private int _outlinedButtonTaps;

    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 10,
            children:
            [
                new Text("Material buttons baseline", fontSize: 20, color: Colors.Black),
                new Text(
                    "TextButton / ElevatedButton / OutlinedButton with enabled/disabled and theme-aware defaults.",
                    fontSize: 14,
                    color: Colors.DimGray),
                new Row(
                    spacing: 8,
                    children:
                    [
                        BuildControlButton(
                            label: _enabled ? "Enabled" : "Disabled",
                            onTap: ToggleEnabled,
                            width: 108,
                            background: Color.Parse("#FFE9F0FF")),
                        BuildControlButton(
                            label: "Reset",
                            onTap: ResetCounters,
                            width: 88,
                            background: Color.Parse("#FFF3E8D8")),
                    ]),
                new Text(
                    $"enabled={_enabled}, text={_textButtonTaps}, elevated={_elevatedButtonTaps}, outlined={_outlinedButtonTaps}",
                    fontSize: 12,
                    color: Colors.DarkSlateGray),
                new SizedBox(
                    width: 240,
                    child: new TextButton(
                        onPressed: _enabled ? OnTextButtonTap : null,
                        child: new Text($"TextButton taps: {_textButtonTaps}"))),
                new SizedBox(
                    width: 240,
                    child: new ElevatedButton(
                        onPressed: _enabled ? OnElevatedButtonTap : null,
                        child: new Text($"ElevatedButton taps: {_elevatedButtonTaps}"))),
                new SizedBox(
                    width: 240,
                    child: new OutlinedButton(
                        onPressed: _enabled ? OnOutlinedButtonTap : null,
                        child: new Text($"OutlinedButton taps: {_outlinedButtonTaps}"))),
                new Row(
                    spacing: 8,
                    children:
                    [
                        new Expanded(
                            child: new ElevatedButton(
                                onPressed: _enabled ? OnElevatedButtonTap : null,
                                backgroundColor: Color.Parse("#FF6A994E"),
                                foregroundColor: Colors.White,
                                child: new Text("Custom elevated"))),
                        new Expanded(
                            child: new OutlinedButton(
                                onPressed: _enabled ? OnOutlinedButtonTap : null,
                                borderColor: Color.Parse("#FF7B2CBF"),
                                foregroundColor: Color.Parse("#FF7B2CBF"),
                                child: new Text("Custom outlined"))),
                    ]),
            ]);
    }

    private Widget BuildControlButton(
        string label,
        Action onTap,
        double width,
        Color background)
    {
        return new SizedBox(
            width: width,
            child: new CounterTapButton(
                label: label,
                onTap: onTap,
                background: background,
                foreground: Colors.Black,
                fontSize: 12,
                padding: new Thickness(10, 8)));
    }

    private void ToggleEnabled()
    {
        SetState(() => _enabled = !_enabled);
    }

    private void ResetCounters()
    {
        SetState(() =>
        {
            _textButtonTaps = 0;
            _elevatedButtonTaps = 0;
            _outlinedButtonTaps = 0;
            _enabled = true;
        });
    }

    private void OnTextButtonTap()
    {
        SetState(() => _textButtonTaps += 1);
    }

    private void OnElevatedButtonTap()
    {
        SetState(() => _elevatedButtonTaps += 1);
    }

    private void OnOutlinedButtonTap()
    {
        SetState(() => _outlinedButtonTaps += 1);
    }
}

File: dart_sample/lib/material_buttons_demo_page.dart
import 'package:flutter/material.dart';

import 'counter_widgets.dart';

class MaterialButtonsDemoPage extends StatefulWidget {
  const MaterialButtonsDemoPage({super.key});

  @override
  State<MaterialButtonsDemoPage> createState() =>
      _MaterialButtonsDemoPageState();
}

class _MaterialButtonsDemoPageState extends State<MaterialButtonsDemoPage> {
  bool _enabled = true;
  int _textButtonTaps = 0;
  int _elevatedButtonTaps = 0;
  int _outlinedButtonTaps = 0;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 10,
      children: <Widget>[
        const Text(
          'Material buttons baseline',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'TextButton / ElevatedButton / OutlinedButton with enabled/disabled and theme-aware defaults.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Row(
          spacing: 8,
          children: <Widget>[
            _buildControlButton(
              label: _enabled ? 'Enabled' : 'Disabled',
              onTap: _toggleEnabled,
              width: 108,
              background: const Color(0xFFE9F0FF),
            ),
            _buildControlButton(
              label: 'Reset',
              onTap: _resetCounters,
              width: 88,
              background: const Color(0xFFF3E8D8),
            ),
          ],
        ),
        Text(
          'enabled=$_enabled, text=$_textButtonTaps, elevated=$_elevatedButtonTaps, outlined=$_outlinedButtonTaps',
          style: const TextStyle(fontSize: 12, color: Colors.blueGrey),
        ),
        SizedBox(
          width: 240,
          child: TextButton(
            onPressed: _enabled ? _onTextButtonTap : null,
            child: Text('TextButton taps: $_textButtonTaps'),
          ),
        ),
        SizedBox(
          width: 240,
          child: ElevatedButton(
            onPressed: _enabled ? _onElevatedButtonTap : null,
            child: Text('ElevatedButton taps: $_elevatedButtonTaps'),
          ),
        ),
        SizedBox(
          width: 240,
          child: OutlinedButton(
            onPressed: _enabled ? _onOutlinedButtonTap : null,
            child: Text('OutlinedButton taps: $_outlinedButtonTaps'),
          ),
        ),
        Row(
          spacing: 8,
          children: <Widget>[
            Expanded(
              child: ElevatedButton(
                onPressed: _enabled ? _onElevatedButtonTap : null,
                style: ElevatedButton.styleFrom(
                  backgroundColor: const Color(0xFF6A994E),
                  foregroundColor: Colors.white,
                ),
                child: const Text('Custom elevated'),
              ),
            ),
            Expanded(
              child: OutlinedButton(
                onPressed: _enabled ? _onOutlinedButtonTap : null,
                style: OutlinedButton.styleFrom(
                  foregroundColor: const Color(0xFF7B2CBF),
                  side: const BorderSide(color: Color(0xFF7B2CBF), width: 1),
                ),
                child: const Text('Custom outlined'),
              ),
            ),
          ],
        ),
      ],
    );
  }

  Widget _buildControlButton({
    required String label,
    required VoidCallback onTap,
    required double width,
    required Color background,
  }) {
    return SizedBox(
      width: width,
      child: CounterTapButton(
        label: label,
        onTap: onTap,
        background: background,
        foreground: Colors.black,
        fontSize: 12,
        padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 8),
      ),
    );
  }

  void _toggleEnabled() {
    setState(() {
      _enabled = !_enabled;
    });
  }

  void _resetCounters() {
    setState(() {
      _textButtonTaps = 0;
      _elevatedButtonTaps = 0;
      _outlinedButtonTaps = 0;
      _enabled = true;
    });
  }

  void _onTextButtonTap() {
    setState(() {
      _textButtonTaps += 1;
    });
  }

  void _onElevatedButtonTap() {
    setState(() {
      _elevatedButtonTaps += 1;
    });
  }

  void _onOutlinedButtonTap() {
    setState(() {
      _outlinedButtonTaps += 1;
    });
  }
}
